server/server.go

```diff
@@ -117,6 +117,11 @@ type Server struct {
 	stats *statistics
 	// GetCertificate is used for loading certificates.
 	GetCertificate GetCertificate
+
+	// UnixListener is the listener serving unix://[UnixAddr]
```

Comment:
It seems to me that we need to take care the unix listener because `defer l.Close()` never gets executed in goroutine when program exits normally. The tcp listener fares better as port will be freed but unix listener will hold the fd.